lib/dirac_clover.cpp

```diff
@@ -59,14 +59,10 @@ namespace quda {
   void DiracClover::MdagM(ColorSpinorField &out, const ColorSpinorField &in) const
   {
     checkFullSpinor(out, in);
+    auto tmp = getTmp(in);
```

Comment:
looking at this now I don't think getTmp with this usage is a good name. It get's some TMP that is stored in an auto variable. How about making it a bit more obvious that we are getting a tmp ColorSpinorField?

Reply:
The object returned here isn't a `ColorSpinorField` and the `auto` was intentional here to avoid having to explicitly state what it was (`FieldTmp<ColorSpinorField>`).

What are your thoughts here?

Reply:
can we rename `getTmp` to `getTmpField` or something like that? Thinking out lout. Just aiming for `the code is the documentation` but otherwise it is not the end of the world as this is not a code touched a lot.

Reply:
I can do that 👍

Reply:
[b37ed336](https://github.com/lattice/quda/commit/b37ed336c69a82e851d2cfccd3ca5472c750f718)